themes/wds_headless/inc/wp-graphql.php

```diff
@@ -238,4 +238,58 @@ function wds_register_archive_seo() {
 		}
 	}
 	add_action( 'graphql_register_types', 'wds_register_archive_seo' );
+
+	/**
+	 * Edit the error messages on user registration.
+	 *
+	 * @author WebDevStudios
+	 * @since 1.0
+	 * @param \WP_Error $errors A WP_Error object containing any errors encountered during registration.
+	 * @return \WP_Error
+	 */
+	function wds_filter_registration_errors( \WP_Error $errors ) {
+		if ( ! $errors->has_errors() ) {
+			return $errors;
+		}
+
+		$new_errors_obj = new \WP_Error();
+		foreach ( $errors->get_error_codes() as $error_code ) {
+			switch ( $error_code ) {
+				case 'empty_username':
+					$error_msg = __( 'Please enter a username.', 'wds-headless-theme' );
+					break;
+				case 'invalid_username':
+					$error_msg = __( 'This username is invalid because it uses illegal characters. Please enter a valid username.', 'wds-headless-theme' );
+					break;
+				case 'username_exists':
+					$error_msg = __( 'This username is already registered. Please choose another one.', 'wds-headless-theme' );
+					break;
+				case 'empty_email':
+					$error_msg = __( 'Please enter your email address.', 'wds-headless-theme' );
+					break;
+				case 'invalid_email':
+					$error_msg = __( 'Please enter a valid email address.', 'wds-headless-theme' );
+					break;
+				case 'email_exists':
+					$error_msg = __( 'This username is already registered. Please choose another one.', 'wds-headless-theme' );
+					break;
+				default:
+					$error_msg = __( 'Registration failed. Please contact the admin.', 'wds-headless-theme' );
```

Comment:
Yes. `esc_html__()` please. Forget `__()` even exists :)

Reply:
Just to add more context why I used `__()` instead of `esc_html__()`. 

The reason why I added this filter is to prevent the escaping of HTML. Because it will be double escaped once its rendered via JSX - https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks

While using `esc_html__()` in this case won't change since its just plain string (no HTML), it kinda defeat the purpose of this filter.

What are your thoughts? @gregrickaby @ravewebdev

Reply:
Thanks for the feedback, but at WDS, we use use `esc_html__()`. Please make the update.

Reply:
Resolved by 80b011ce3f11da6f76cdd61894da54a93fe8948a